### PR TITLE
Add meta tag to attachment to show images in mastodon app.

### LINKED ISSUE #12197: Fill meta.original.[width/height] keys in ActivityPub media attachments
@j-f1 wrote in https://github.com/mastodon/mastodon-ios/issues/310#issuecomment-1318050979 :
> The underlying issue here is that the Mastodon app requires `meta.original.[width/height]` to be provided so it can lay out its UI appropriately. Any media attachments that don’t provide that information will be silently dropped, for better or for worse. The rest of the properties are optional, as far as I can tell.

### PR DESCRIPTION
**Pull request by Schnoop**
Fix that images are not shown in Mastodon app.

https://github.com/friendica/friendica/issues/12197

### CODE REVIEW

File: src/App/Page.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ public function logRuntime(IManageConfigValues $config, string $origin = '')
 	 * The return value will be casted to boolean if non-boolean was returned.
 	 * @since 5.0.0
 	 */
-	public function offsetExists($offset): bool
+	public function offsetExists(mixed $offset): bool
```
Comment (MrPetovan) [inline src/App/Page.php]: Thanks for the edit, unfortunately the minimum PHP version we support is still PHP 7.3 (per `composer.json`) where this type-hint doesn't exist yet. Please remove any other references to this type.

File: src/Content/Nav.php [hunk 2]
```diff
@@ -245,7 +245,7 @@ private static function getInfo(App $a): array
 
 		$gdirpath = 'directory';
 
-		if (strlen(DI::config()->get('system', 'singleuser'))) {
+		if (empty(DI::config()->get('system', 'singleuser'))) {
```
Comment (MrPetovan) [inline src/Content/Nav.php]: `empty` has the opposite logical meaning as `strlen` in this case. Please fix using either suggestion:
```suggestion
		if (!empty(DI::config()->get('system', 'singleuser'))) {
```
```suggestion
		if (DI::config()->get('system', 'singleuser') ?? '') {
```

File: src/Core/Session/Handler/Database.php [hunk 3]
```diff
@@ -57,7 +57,7 @@ public function open($path, $name): bool
 		return true;
 	}
 
-	public function read($id)
+	public function read($id): string|false
```
Comment (MrPetovan) [inline src/Core/Session/Handler/Database.php]: Union types aren't supported in PHP 7.3 yet.

File: src/Model/Log/ParsedLogIterator.php [hunk 4]
```diff
@@ -202,7 +202,7 @@ public function key(): int
 	 * @see Iterator::current()
 	 * @return ?ParsedLogLing
 	 */
-	public function current()
+	public function current(): mixed
```
Comment (MrPetovan) [inline src/Model/Log/ParsedLogIterator.php]: Please remove this type-hint as well.

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Thank you for your work, sorry we don't support PHP 8 minimum yet!

**Review by Schnoop — COMMENTED** [review]
Reverted changes that collides with PHP 7.3

**Comment by MrPetovan** [comment]
Thanks, not all the changes needed to be reverted, we still are aiming for PHP 8 compatibility at some point, and you found several places where I expect PHP 8 would issue a deprecation warning/type error? It was valuable. Do you mind reintroducing them? Alternatively, I can do it myself.

**Comment by Schnoop** [comment]
Let's have a separate Pull request for those changes. There are a lot of things to fix for PHP 8.x. Some of the vendor libraries are not ready for PHP 8.x too.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**